tools/osv-scanner/main.go

```diff
@@ -37,12 +37,15 @@ func scanDir(query *osv.BatchedQuery, dir string) error {
 
 		if !info.IsDir() {
 			if parser, _ := lockfile.FindParser(path, ""); parser != nil {
-				scanLockfile(query, path)
+				err := scanLockfile(query, path)
 				if err != nil {
 					log.Println("Attempted to scan lockfile but failed: " + path)
 				}
 			}
-			scanSBOMFile(query, path)
+			extension := strings.ToLower(filepath.Ext(info.Name()))
+			if extension == ".json" || strings.Contains(info.Name(), ".spdx") {
+				_ = scanSBOMFile(query, path)
```

Comment:
What's the purpose of performing this scan if not checking the error?

Reply:
If this succeeds then the dependencies are added to the `query` array. If this fails, then this is not an SBOM file, so nothing happens.